src/prover_disk.hpp

```diff
@@ -635,11 +639,18 @@ class DiskProver {
     // all of the leaves (x values). For example, for depth=5, it fetches the position-th
     // entry in table 5, reading the two back pointers from the line point, and then
     // recursively calling GetInputs for table 4.
-    std::vector<Bits> GetInputs(uint64_t position, uint8_t depth)
+    std::vector<Bits> GetInputs(uint64_t position, uint8_t depth, std::ifstream* disk_file=nullptr)
     {
-        // Create individual file handles to allow parallel processing
-        std::ifstream disk_file(filename, std::ios::in | std::ios::binary);
-        uint128_t line_point = ReadLinePoint(disk_file, depth, position);
+        uint128_t line_point;
+
+        if (!disk_file) {
+            // No disk file passed in, so we assume here we are doing parallel reads
+            // Create individual file handles to allow parallel processing
+            std::ifstream disk_file(filename, std::ios::in | std::ios::binary);
+            line_point = ReadLinePoint(disk_file, depth, position);
```

Comment:
```suggestion
            std::ifstream disk_file_parallel(filename, std::ios::in | std::ios::binary);
            line_point = ReadLinePoint(disk_file_parallel, depth, position);
```

To not shadow the parameter